api/filter/network/http_connection_manager.proto

```diff
@@ -80,6 +80,23 @@ message HttpConnectionManager {
     // populate the tag name, and the header value is used to populate the tag value. The tag is
     // created if the specified header name is present in the request's headers.
     repeated string request_headers_for_tags = 2;
+
+    // Target percentage of requests managed by this HTTP connectionmanager that will be force
+    // traced if the *x-client-trace-id* header is set. Defaults to 100%. This variable is a direct
+    // analog for the variable of the same name in the :ref:`HTTP Connection Manager
+    // <config_http_conn_man_runtime>`.
+    double client_sampling = 3 [(validate.rules).double = {gte: 0, lte: 100}];
```

Comment:
I think we should do the following:
* Add a universal `Percent` message, which enforced [0..100] range and is backed by `double`.
* Express locally, at each field, what the actual effective interpretation will be. This interpretation might change over time (e.g. it might be rounded to nearest 1% today, in later Envoy releases it might be 0.1%). I agree with @wora that folks supplying the API data should be able to express their intents as fine grained as they wish, but at the same time, users need to know what the limits at the implementation level are.